### PR TITLE
Activate CSP

### PR DESCRIPTION
**Pull request by NicolasDorier**
Attempt to activate some limited CSP to prevent XSS vulnerabilities to escalate too much.
This also fix a bug with the previous root domain mapping, where HTTP headers were not properly set if using root domain mapping to a crowdfund or pos app.

### CODE REVIEW

File: BTCPayServer/Controllers/AppsPublicController.cs [hunk 1]
```diff
@@ -104,10 +107,13 @@ public async Task<IActionResult> ViewPointOfSale(string appId, PosViewType? view
         }
 
         [HttpPost]
+        [ContentSecurityPolicyAttribute(Enabled = false)]
+        [Route("/")]
```
Comment (Kukks) [inline BTCPayServer/Controllers/AppsPublicController.cs:111]: This route set on all actions might cause MVC to crash due to a collision

Comment (NicolasDorier) [reply BTCPayServer/Controllers/AppsPublicController.cs:111]: no it does not. Because we use `DomainMappingConstraint`

Comment (NicolasDorier) [reply BTCPayServer/Controllers/AppsPublicController.cs:111]: MVC select the route by enumerating all, and for each route calculating the constraints. Problem arise if after applying the constraints, there is more than one route.

In this case, this is not the case.

**Comment by NicolasDorier** [comment]
I will not merge this, this does not protect agaisnt XSS and risk of breaking too many things.

**Comment by NicolasDorier** [comment]
I will probably take parts of this PR though.

**Comment by pavlenex** [comment]
@NicolasDorier is this one ready to be closed then?

**Comment by NicolasDorier** [comment]
yeah... might revist later, but this PR might introduce bugs and does not really protect everything...

**Pull request closed without merging**